pkg/apis/serving/k8s_validation.go

```diff
@@ -609,3 +619,28 @@ func ValidateNamespacedObjectReference(p *corev1.ObjectReference) *apis.FieldErr
 	}
 	return errs
 }
+
+// This is attached to contexts as they are passed down through a user container
+// being validated.
+type userContainer struct{}
+
+// WithUserContainer notes on the context that further validation or defaulting
+// is within the context of a user container in the revision.
+func WithinUserContainer(ctx context.Context) context.Context {
+	return context.WithValue(ctx, userContainer{}, struct{}{})
+}
+
+// This is attached to contexts as they are passed down through a sidecar container
+// being validated.
+type sidecarContainer struct{}
+
+// WithinSidecatrContainer notes on the context that further validation or defaulting
+// is within the context of a sidecar container in the revision.
+func WithinSidecarContainer(ctx context.Context) context.Context {
+	return context.WithValue(ctx, sidecarContainer{}, struct{}{})
+}
+
+// Check if we are in the context of a sidecar container in the revision.
+func IsInSidecarContainer(ctx context.Context) bool {
+	return ctx.Value(sidecarContainer{}) != nil
+}
```

Comment:
There might be a better place for the helpers but they are also specific to this file for now.